spark-connect/server/src/main/scala-spark-master/io/delta/connect/DeltaRelationPlugin.scala

```diff
@@ -275,6 +277,21 @@ class DeltaRelationPlugin extends RelationPlugin with DeltaPlannerBase {
       key = planner.transformExpression(assignment.getField),
       value = planner.transformExpression(assignment.getValue))
   }
+
+  private def transformOptimizeTable(
+      planner: SparkConnectPlanner, optimizeTable: proto.OptimizeTable): LogicalPlan = {
+    val deltaTable = transformDeltaTable(planner, optimizeTable.getTable)
+    var optimizeBuilder = deltaTable.optimize()
+    for (partitionFilter <- optimizeTable.getPartitionFiltersList.asScala) {
+      optimizeBuilder = optimizeBuilder.where(partitionFilter)
+    }
+    val df = if (optimizeTable.getZorderColumnsList.isEmpty) {
+      optimizeBuilder.executeCompaction()
+    } else {
+      optimizeBuilder.executeZOrderBy(optimizeTable.getZorderColumnsList.asScala.toSeq: _*)
+    }
+    df.queryExecution.commandExecuted
```

Comment:
This should not be needed if the plan produced by optimize is a Command.

Reply:
better safe than sorry I suppose